src/dfm/cleaning/quality_filter.py

```diff
@@ -0,0 +1,697 @@
+            n-grams range top_ngram_chr_fraction_range. I.e. [0.20, 0.18, 0.16] with a
+            range of (2, 4) states that if the first top 20% of characters is contained
+            within the top 2-gram then filter out the text.
+        top_ngram_chr_fraction_range (Tuple[int, int], optional): Range of n-gram to
+            check for top_ngram_chr_fraction_thresholds.
+        duplicate_n_gram_fraction_thresholds (List[float], optional): The character
+            fraction thresholds. Defaults to [0.15, 0.14, 0.13, 0.12, 0.11, 0.10],
+            which for example denote that the any text with duplicate 5 grams
+            constituting more than 15% of the text characters is filtered, 14% for
+            6-grams and so on.
+        duplicate_n_gram_fraction_range (Tuple[int, int], optional): The n-gram range.
+            Defaults to (5, 11).
+        max_length (int, optional): max_length in characters
+        string (str, optional): String for filtering. Defaults to None.
+    """
+
+    def __init__(
+        self,
+        stop_words: Optional[Set[str]] = None,
+        min_stop_words: int = 2,
+        mean_word_length: Tuple[int, int] = (3, 10),
+        doc_length: Tuple[int, int] = (50, 100_000),
+        alpha_ratio: float = 0.7,
+        symbol_2_word_hashtag: float = 0.1,
+        symbol_2_word_ellipsis: float = 0.1,
+        max_p_begin_bullets: float = 0.9,
+        max_p_end_ellipsis: float = 0.3,
+        duplicate_line_fraction: float = 0.3,
+        duplicate_paragraph_fraction: float = 0.3,
+        duplicate_lines_chr_fraction: float = 0.2,
+        duplicate_paragraph_chr_fraction: float = 0.2,
+        top_ngram_chr_fraction_thresholds: List[float] = [0.20, 0.18, 0.16],
+        top_ngram_chr_fraction_range: Tuple[int, int] = (2, 4),
+        duplicate_n_gram_fraction_thresholds: List[float] = [
+            0.15,
+            0.14,
+            0.13,
+            0.12,
+            0.11,
+            0.10,
+        ],
+        duplicate_n_gram_fraction_range: Tuple[int, int] = (5, 10),
+        max_length: int = 5_000_000,
+        string_filter: Optional[str] = None,
+    ):
+        if stop_words is None:
+            stop_words = set(
+                [
+                    "er",
+                    "jeg",
+                    "det",
+                    "du",
+                    "ikke",
+                    "at",
+                    "en",
+                    "og",
+                    "har",
+                    "vi",
+                    "til",
+                    "på",
+                    "hvad",
+                    "mig",
+                    "med",
+                    "de",
+                    "for",
+                    "den",
+                    "så",
+                    "der",
+                    "dig",
+                    "han",
+                    "kan",
+                    "af",
+                ]
+            )
+
+        self.nlp = spacy.blank("da")
+
+        self.__set_getters()
```

Comment:
What is this doing? Can you add a comment?

Reply:
better?